packages/headless/src/controllers/commerce/context/headless-context.ts

```diff
@@ -0,0 +1,187 @@
+import {RecordValue, Schema } from "@coveo/bueno";
+import {nonEmptyString, requiredNonEmptyString, validateOptions} from '../../../utils/validate-payload';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../controller/headless-controller';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  setClientId,
+  setContext,
+  setCurrency,
+  setLanguage,
+  setTrackingId,
+  setUser, setView
+} from '../../../features/commerce/context/context-actions';
+import {contextReducer as commerceContext} from '../../../features/commerce/context/context-slice';
+
+const optionsSchema = new Schema({
+  trackingId: nonEmptyString,
+  language: nonEmptyString,
+  currency: nonEmptyString,
+  clientId: nonEmptyString,
+  user: new RecordValue({
+    values: {
+      userId: nonEmptyString,
+      email: nonEmptyString,
+      userIp: nonEmptyString,
+      userAgent: nonEmptyString,
+    },
+  }),
+  view: new RecordValue({
+    options: { required: true },
+    values: {
+      url: requiredNonEmptyString,
+    }
+  })
+});
+
+export interface ContextOptions {
+  trackingId?: string;
+  language?: string;
+  currency?: string;
+  clientId?: string;
+  user?: User;
+  view: View;
+}
+
+export interface User {
```

Comment:
Kind of similar as my above comment: In general, interface where all properties are optional makes me suspicious 😉 

Wouldn't a `userId` always be required ?

Reply:
Great point. I've relayed this question to the team channel, as it's not something the API makes explicit: https://coveo.slack.com/archives/C05FR3KKBHB/p1695913390115019

Reply:
We'll require either a userId or an email. I'll adjust the PR accordingly!

Reply:
Done https://github.com/coveo/ui-kit/pull/3190/commits/eff279c4b77d2f02990b2ae2d5848240b24beb8a